inc/class-yoast-acf-analysis.php

```diff
@@ -30,69 +30,65 @@ public function admin_init() {
 
 		$this->boot();
 
-			if ( defined( 'YOAST_ACF_ANALYSIS_ENVIRONMENT' ) && 'development' === YOAST_ACF_ANALYSIS_ENVIRONMENT ) {
-				$this->boot_dev();
-			}
+		if ( defined( 'YOAST_ACF_ANALYSIS_ENVIRONMENT' ) && 'development' === YOAST_ACF_ANALYSIS_ENVIRONMENT ) {
+			$this->boot_dev();
 		}
 
+		$this->register_config_filters();
+
 		$assets = new Yoast_ACF_Analysis_Assets();
 		$assets->init();
 	}
 
 	/**
-	 * Boot the plugin
+	 * Boots the plugin.
 	 */
 	public function boot() {
 
-		if ( is_null( Yoast_ACF_Analysis_Registry::instance()->get( 'config' ) ) ) {
-
-			$default_configuration = new Yoast_ACF_Analysis_Configuration_Default( $this->get_blacklist(), $this->get_field_selectors() );
-
-			$configuration = apply_filters(
-				Yoast_ACF_Analysis_Configuration::PLUGIN_NAME . '/config',
-				$default_configuration
-			);
-
-			if ( ! ($configuration instanceof Yoast_ACF_Analysis_Configuration) ) {
-				$configuration = $default_configuration;
-			}
-
-			Yoast_ACF_Analysis_Registry::instance()->add( 'config', $configuration );
-
+		$registry = Yoast_ACF_Analysis_Facade::get_registry();
+		if ( null !== $registry->get( 'config' ) ) {
```

Comment:
I read this message yesterday and let it sink in a bit more. Realizing that you come from an agency perspective the usecase to have this functionality is very clear.

Looking at the things that can be filtered, I would think that the specific modules would be the only thing that might change between projects, thus having to implement the entire interface to provide a custom configuration would be a lot of overhead.

What do you think about just filtering the class and thus providing the ability to extend the class instead of having to implement the entire interface?